### PR TITLE
Remove `gtk` builds from CI and releases, change default build to no gui, remove deprecated env vars from `justfile`.

### PR DESCRIPTION
**Pull request by ImperatorStorm**

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -33,70 +33,70 @@ build-win-gnu:
...
-    cargo build --target=x86_64-unknown-linux-gnu --release --no-default-features
+    CC=clang-14 cargo rustc --target=x86_64-unknown-linux-gnu --release --no-default-features -- -Clink-self-contained=yes -Clinker=rust-lld
     zip -r out/ferium-linux-gnu-nogui.zip -j target/x86_64-unknown-linux-gnu/release/ferium
 
 # Build for GNU Linux ARM64 with a GTK backend
 build-linux-arm64-gtk:
     rm -f out/ferium-linux-gnu-arm64-gtk.zip
     mkdir -p out
-    CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-linux-gnu-gcc cargo build --target=aarch64-unknown-linux-gnu --release
+    CC=clang-14 cargo rustc --target=aarch64-unknown-linux-gnu --release --features gtk -Clink-self-contained=yes -Clinker=rust-lld
```
Comment (theofficialgman) [inline justfile:57]: you fogot a `--` in this command as well as multiple others

**Comment by theofficialgman** [comment]
also since you have switched the justfile to use clang exclusively. you should probably also make the `tests` use clang as well in the yml

**Comment by theofficialgman** [comment]
tested the arm64 xdg build, works fine for me

I tried building the GTK version ` CC=clang cargo rustc --target=aarch64-unknown-linux-gnu --release --features gtk -- -Clink-self-contained=yes -Clinker=rust-lld` on native arm64 and go the following errors
```
   Compiling ferium v4.1.10 (/tmp/ferium)
error: linking with `rust-lld` failed: exit status: 1
  |
  = note: "rust-lld" "-flavor" "gnu" "/tmp/rustcZNSSV8/symbols.o" "/tmp/ferium/target/aarch64-unknown-linux-gnu/release/deps/ferium-b91d8d8305a0537d.ferium.72da06f8-cgu.0.rcgu.o" "--as-needed" "-L" "/tmp/ferium/target/aarch64-unknown-linux-gnu/release/deps" "-L" "/tmp/ferium/target/release/deps" "-L" "/tmp/ferium/target/aarch64-unknown-linux-gnu/release/build/ring-1850fbbe9c6e59df/out" "-L" "/usr/local/lib" "-L" "/usr/local/lib" "-L" "/tmp/ferium/target/aarch64-unknown-linux-gnu/release/build/bzip2-sys-f68a203ea2f434b3/out/lib" "-L" "/tmp/ferium/target/aarch64-unknown-linux-gnu/release/build/zstd-sys-e51c6eebe26b5569/out" "-L" "/home/garrett/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "-Bstatic" "/tmp/rustcZNSSV8/libzstd_sys-b688b2e42841b250.rlib" "/tmp/rustcZNSSV8/libbzip2_sys-fb05834317a94d61.rlib" "/tmp/rustcZNSSV8/libring-07a7117292f05f5f.rlib" "--start-group" "--end-group" "/home/garrett/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcompiler_builtins-e528c2f86745b3a4.rlib" "-Bdynamic" "-lgtk-3" "-lgdk-3" "-lpangocairo-1.0" "-lpango-1.0" "-latk-1.0" "-lcairo-gobject" "-lcairo" "-lgdk_pixbuf-2.0" "-lgio-2.0" "-lgobject-2.0" "-lglib-2.0" "-latk-1.0" "-lgobject-2.0" "-lglib-2.0" "-lgdk-3" "-lpangocairo-1.0" "-lpango-1.0" "-lgdk_pixbuf-2.0" "-lcairo-gobject" "-lcairo" "-lgobject-2.0" "-lglib-2.0" "-lpango-1.0" "-lgobject-2.0" "-lglib-2.0" "-lgdk_pixbuf-2.0" "-lgobject-2.0" "-lglib-2.0" "-lgio-2.0" "-lgobject-2.0" "-lglib-2.0" "-lcairo" "-lgobject-2.0" "-lglib-2.0" "-lgobject-2.0" "-lglib-2.0" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "--eh-frame-hdr" "-znoexecstack" "-L" "/home/garrett/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "-L" "/home/garrett/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/self-contained" "-o" "/tmp/ferium/target/aarch64-unknown-linux-gnu/release/deps/ferium-b91d8d8305a0537d" "--gc-sections" "-pie" "-zrelro" "-znow" "-O1" "--strip-all"
  = note: rust-lld: error: unable to find library -lpangocairo-1.0
          rust-lld: error: unable to find library -lpango-1.0
          rust-lld: error: unable to find library -latk-1.0
          rust-lld: error: unable to find library -lcairo-gobject
          rust-lld: error: unable to find library -lcairo
          rust-lld: error: unable to find library -lgdk_pixbuf-2.0
          rust-lld: error: unable to find library -lgio-2.0
          rust-lld: error: unable to find library -lgobject-2.0
          rust-lld: error: unable to find library -lglib-2.0
          rust-lld: error: unable to find library -latk-1.0
          rust-lld: error: unable to find library -lgobject-2.0
          rust-lld: error: unable to find library -lglib-2.0
          rust-lld: error: unable to find library -lpangocairo-1.0
          rust-lld: error: unable to find library -lpango-1.0
          rust-lld: error: unable to find library -lgdk_pixbuf-2.0
          rust-lld: error: unable to find library -lcairo-gobject
          rust-lld: error: unable to find library -lcairo
          rust-lld: error: unable to find library -lgobject-2.0
          rust-lld: error: unable to find library -lglib-2.0
          rust-lld: error: unable to find library -lpango-1.0
          rust-lld: error: too many errors emitted, stopping now (use -error-limit=0 to see all errors)
          

error: could not compile `ferium` due to previous error
```
(yes I have the dev libraries already installed)

**Comment by ImperatorStorm** [comment]
> also since you have switched the justfile to use clang exclusively. you should probably also make the `tests` use clang as well in the build.yml

This should be resolved.

**Comment by ImperatorStorm** [comment]
> https://github.com/gorilla-devs/ferium/pull/197#issuecomment-1229011751

What OS did you try this on?

**Comment by theofficialgman** [comment]
I was using my bionic 18.04 arm64 install to test

the reality is, if we don't plan on using it in the justfile for the CI, it is fine for it to use GCC and dynamic linking, since the users of it will be building on their system for their system, so there is no downside to dynamic linking.

also it is not ARM64 clang that is cursed. clang with the rust-lld linker just doesn't work with gtk. same issue on x86_64 
` CC=clang-14 cargo rustc --target=x86_64-unknown-linux-gnu --release --features gtk -- -Clink-self-contained=yes -Clinker=rust-lld`
```
error: linking with `rust-lld` failed: exit status: 1
  |
  = note: "rust-lld" "-flavor" "gnu" "/tmp/rustc0wj2Tq/symbols.o" "/tmp/ferium/target/x86_64-unknown-linux-gnu/release/deps/ferium-f77bc2861caff437.ferium.b263d0b7-cgu.0.rcgu.o" "--as-needed" "-L" "/tmp/ferium/target/x86_64-unknown-linux-gnu/release/deps" "-L" "/tmp/ferium/target/release/deps" "-L" "/tmp/ferium/target/x86_64-unknown-linux-gnu/release/build/ring-99a645e84d3d5a7a/out" "-L" "/tmp/ferium/target/x86_64-unknown-linux-gnu/release/build/bzip2-sys-5b68b10fce4d1695/out/lib" "-L" "/tmp/ferium/target/x86_64-unknown-linux-gnu/release/build/zstd-sys-9ffbeb1dcdf8fb18/out" "-L" "/home/garrett/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Bstatic" "/tmp/rustc0wj2Tq/libzstd_sys-fec02bd3653c7431.rlib" "/tmp/rustc0wj2Tq/libbzip2_sys-f741169b8a8deab4.rlib" "/tmp/rustc0wj2Tq/libring-b65f2d6cb4f671e5.rlib" "--start-group" "--end-group" "/home/garrett/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-c21be34a5cae8449.rlib" "-Bdynamic" "-lgtk-3" "-lgdk-3" "-lpangocairo-1.0" "-lpango-1.0" "-lharfbuzz" "-latk-1.0" "-lcairo-gobject" "-lcairo" "-lgdk_pixbuf-2.0" "-lgio-2.0" "-lgobject-2.0" "-lglib-2.0" "-latk-1.0" "-lgobject-2.0" "-lglib-2.0" "-lgdk-3" "-lpangocairo-1.0" "-lpango-1.0" "-lharfbuzz" "-lgdk_pixbuf-2.0" "-lcairo-gobject" "-lcairo" "-lgobject-2.0" "-lglib-2.0" "-lpango-1.0" "-lgobject-2.0" "-lglib-2.0" "-lharfbuzz" "-lgdk_pixbuf-2.0" "-lgobject-2.0" "-lglib-2.0" "-lgio-2.0" "-lgobject-2.0" "-lglib-2.0" "-lcairo" "-lgobject-2.0" "-lglib-2.0" "-lgobject-2.0" "-lglib-2.0" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "--eh-frame-hdr" "-znoexecstack" "-L" "/home/garrett/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-L" "/home/garrett/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/self-contained" "-o" "/tmp/ferium/target/x86_64-unknown-linux-gnu/release/deps/ferium-f77bc2861caff437" "--gc-sections" "-pie" "-zrelro" "-znow" "-O1" "--strip-all"
  = note: rust-lld: error: unable to find library -lgtk-3
          rust-lld: error: unable to find library -lgdk-3
          rust-lld: error: unable to find library -lpangocairo-1.0
          rust-lld: error: unable to find library -lpango-1.0
          rust-lld: error: unable to find library -lharfbuzz
          rust-lld: error: unable to find library -latk-1.0
          rust-lld: error: unable to find library -lcairo-gobject
          rust-lld: error: unable to find library -lcairo
          rust-lld: error: unable to find library -lgdk_pixbuf-2.0
          rust-lld: error: unable to find library -lgio-2.0
          rust-lld: error: unable to find library -lgobject-2.0
          rust-lld: error: unable to find library -lglib-2.0
          rust-lld: error: unable to find library -latk-1.0
          rust-lld: error: unable to find library -lgobject-2.0
          rust-lld: error: unable to find library -lglib-2.0
          rust-lld: error: unable to find library -lgdk-3
          rust-lld: error: unable to find library -lpangocairo-1.0
          rust-lld: error: unable to find library -lpango-1.0
          rust-lld: error: unable to find library -lharfbuzz
          rust-lld: error: unable to find library -lgdk_pixbuf-2.0
          rust-lld: error: too many errors emitted, stopping now (use -error-limit=0 to see all errors)
```

just leave the justfile unchanged for GTK. let it use dynamic linking and build with GCC like before.

`cargo rustc --target=x86_64-unknown-linux-gnu --release --features gtk --`
or `cargo build --target=x86_64-unknown-linux-gnu --release`

**Comment by ImperatorStorm** [comment]
> https://github.com/gorilla-devs/ferium/pull/197#issuecomment-1229343111

It took me 4 hours to figure it out, but I fixed it.

`rust-lld` seems to just be broken, and I could reproduce the error with `-Clinker=lld`. I worked around this by having rust invoke `clang` as linker, and having `clang` invoke `lld` via `LDFLAGS=lld`.

**Comment by theofficialgman** [comment]
I believe rust-lld must be used when cross compiling (hence why the arm64 in your github actions are now failing, gnu and musl no longer work). 

honestly I would just leave this alone. what is there currently works, even if it might look a little ugly.
you are welcome to keep tyring to find something that works... but it might be a waste of your valuable time.

**Comment by ImperatorStorm** [comment]
> I believe rust-lld must be used when cross compiling (hence why the arm64 in your github actions are now failing, gnu and musl no longer work).
> 
> honestly I would just leave this alone. what is there currently works, even if it might look a little ugly. you are welcome to keep tyring to find something that works... but it might be a waste of your valuable time.

fixed

**Comment by ImperatorStorm** [comment]
Ok, I give up. How it works is beyond my level of comprehension.

I did clean up `justfile`, however.

**Comment by theofficialgman** [comment]
I have verified the arm64 binaries still work and approve these changes. ready to merge

**Pull request merged**